File: leetcode/0001_0099/0001.py
from typing import List


class Solution:
    """
    See: https://leetcode.cn/problems/two-sum/description/
    """

    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i, num in enumerate(nums):
            if map.get(target - num, -1) != -1:
                return [i, map[target - num]]
            map[num] = i
        return [-1, -1]

File: leetcode/0001_0099/0001.js
/**
 *
 * @see https://leetcode.cn/problems/two-sum/description/
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const store = {};
  for (let i = 0; i < nums.length; i++) {
    if (store[target - nums[i]] !== void 0) return [i, store[target - nums[i]]]
    store[nums[i]] = i
  }
  return [-1, -1]
};
